Feature/metadata push

jofsky opened the pull request:
# Critical Changes

# Changes

# Issues Closed

jofsky commented:
This task will be used to update packagepusherrors from a packaging org.
To run the task requires the following command:

`cci task run meta_push`

This task takes all package push errors and inserts it into the appropriate table in heroku postgres. Then a view is created in heroku postgres to join all the tables with the same schema together to make meaningful relations. As a result we will be able to see the org and error that was associated with it in close to live time. Still a couple kinks in reading the format for time, and need to figure out how to make it schema name agnostic without explicitly passing in the value.

davisagli commented:
@jofsky What sort of feedback would be helpful to you at this stage? This doesn't look ready to merge; some things I noticed are:

- pushupgrades.sql isn't used anywhere; I expected to find at least see some instructions for when or how to use it manually
- the task uses a metapush_postgres service to get the database url but that service isn't defined anywhere (it probably isn't necessary to use a CCI service anyway; DATABASE_URL is available as an environment variable on Heroku)
- the task isn't actually inserting anything into the database

I'm also kind of forgetting the reason we need to do this rather than just getting the package push errors via Heroku Connect. Was it an issue with syncing that particular table using Heroku Connect? This would be a good thing to document both in the PR notes and in a comment in the code, so that we have a trail to follow to check whether the limitation is still there in the future.

We probably need to schedule some time with Jason once he's back to make sure we're on the same page about how this will be used:
- How will we set this up to run periodically against multiple packaging orgs
- How will the Heroku app authenticate to the packaging orgs

jofsky commented:
So I purposefully commented out the execute command since its live pushing to our tables. I fixed the timing issue last night after commenting on it since it was bothering me but is now fixed. I do agree that we need to meet with jason about design itself whether we want to modularize the data based on schema name. Additionally the function written in sql isn't used because its already defined in the database and the view already exists, the function updates the view each time.
